src/webui/api/apierror.h

```diff
@@ -26,26 +26,26 @@
  * exception statement from your version.
  */
 
-#ifndef JSONUTILS_H
-#define JSONUTILS_H
+#pragma once
 
-#include <QVariant>
-#include <QJsonDocument>
-#include <QJsonObject>
-#include <QJsonArray>
+#include "base/exceptions.h"
 
-namespace json {
-
-    inline QByteArray toJson(const QVariant& var)
+class APIError : public RuntimeError
+{
+public:
+    enum Type
```

Comment:
`enum class`?

Reply:
I don't see much sense to have a separate namespace in this case (this type isn't shared to have it outside APIError; having names like APIError::ErrorType::NotFound seems too complex for me). Have you serious reasons to change it?

Reply:
>Have you serious reasons to change it?

Nothing much except modern practice and some type safety/correctness...
Personally I would prefer `enum class` whenever possible.

Reply:
Ok, use `enum class` now.